sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -0,0 +1,181 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.os.Bundle;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.Breadcrumb;
+import io.sentry.IHub;
+import io.sentry.ITransaction;
+import io.sentry.Integration;
+import io.sentry.SentryLevel;
+import io.sentry.SentryOptions;
+import io.sentry.SpanStatus;
+import io.sentry.util.Objects;
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.WeakHashMap;
+import org.jetbrains.annotations.NotNull;
+
+public final class ActivityLifecycleIntegration
+    implements Integration, Closeable, Application.ActivityLifecycleCallbacks {
+
+  private final @NotNull Application application;
+  private @NotNull IHub hub;
+  private @NotNull SentryAndroidOptions options;
+
+  private boolean performanceEnabled = false;
+
+  // WeakHashMap isn't thread safe but ActivityLifecycleCallbacks is only called from the
+  // main-thread
+  private final @NotNull WeakHashMap<Activity, ITransaction> activities = new WeakHashMap<>();
```

Comment:
Could have a more descriptive name:

```suggestion
  private final @NotNull WeakHashMap<Activity, ITransaction> activitiesWithOngoingTransactions = new WeakHashMap<>();
```

^ suggestions is maybe too descriptive but the goal is to describe what this is about